components/event-publisher-proxy/pkg/cloudevents/builder/generic_test.go

```diff
@@ -0,0 +1,224 @@
+			givenSource:          "source1",
+			givenType:            "o-rder.creat ed.v1",
+			givenApplicationName: "appName1",
+			wantType:             "prefix.source1.o-rder.created.v1",
+			wantSource:           "source1",
+		},
+		{
+			name:                 "should return application name as source",
+			givenSource:          "appName1",
+			givenType:            "order.created.v1",
+			givenApplicationName: "appName1",
+			wantType:             "prefix.appName1.order.created.v1",
+			wantSource:           "appName1",
+		},
+		{
+			name:                   "should return application label as source",
+			givenSource:            "appName1",
+			givenType:              "order.created.v1",
+			givenApplicationName:   "appName1",
+			givenApplicationLabels: map[string]string{application.TypeLabel: "t..e--s__t!!a@@p##p%%t^^y&&p**e"},
+			wantType:               "prefix.testapptype.order.created.v1",
+			wantSource:             "testapptype",
+		},
+		{
+			name:                 "should return error if empty type",
+			givenSource:          "source1",
+			givenType:            "",
+			givenApplicationName: "appName1",
+			wantError:            true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// given
+			// build cloud event
+			builder := testingutils.NewCloudEventBuilder(
+				testingutils.WithCloudEventSource(tc.givenSource),
+				testingutils.WithCloudEventType(tc.givenType),
+			)
+			payload, _ := builder.BuildStructured()
+			newEvent := cloudevents.NewEvent()
+			newEvent.SetType(testingutils.CloudEventTypeWithPrefix)
+			err := json.Unmarshal([]byte(payload), &newEvent)
+			require.NoError(t, err)
+
+			appLister := fake.NewApplicationListerOrDie(
+				context.Background(),
+				applicationtest.NewApplication(tc.givenApplicationName, tc.givenApplicationLabels))
+
+			genericBuilder := &GenericBuilder{
+				typePrefix:        "prefix",
+				applicationLister: appLister,
+				logger:            logger,
+				cleaner:           cleaner.NewJetStreamCleaner(logger),
+			}
+
+			// when
+			buildEvent, err := genericBuilder.Build(newEvent)
+
+			// then
+			if tc.wantError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tc.wantSource, buildEvent.Source())
+				require.Equal(t, tc.wantType, buildEvent.Type())
+			}
+		})
+	}
+}
+
+func Test_GetAppNameOrSource(t *testing.T) {
+	t.Parallel()
+
+	// init the logger
+	logger, err := kymalogger.New("json", "debug")
+	if err != nil {
+		golog.Fatalf("Failed to initialize logger, error: %v", err)
+	}
+
+	testCases := []struct {
+		name                   string
+		givenApplicationName   string
+		givenApplicationLabels map[string]string
+		givenSource            string
+		wantSource             string
+	}{
+		{
+			name:                 "should return application name instead of source name",
+			givenSource:          "appName1",
```

Comment:
given source name should be different than given app name, should it not be?

Reply:
No, its needs to be the same, because we use the value of givenSource to find any application by that name, so inshort applicationName = sourceName.